framework/audio/isequencer.h

```diff
@@ -0,0 +1,77 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_ISEQUENCER_H
+#define MU_AUDIO_ISEQUENCER_H
+
+#include <variant>
+
+#include "async/channel.h"
+#include "modularity/imoduleexport.h"
+#include "iaudioplayer.h"
+#include "imidiplayer.h"
+#include "iaudiostream.h"
+namespace mu {
+namespace audio {
+class ISequencer : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(ISequencer)
+
+public:
+    virtual ~ISequencer() = default;
+
+    using midi_track_t = std::shared_ptr<IMIDIPlayer>;
+    using audio_track_t = std::shared_ptr<IAudioPlayer>;
+    using track_t = std::shared_ptr<IPlayer>;
```

Comment:
Usually "_t" are simple types or structures, it is very unexpected to find that these are smart pointers to classes

Reply:
fixed https://github.com/musescore/MuseScore/pull/7274